File: easy/buzz_lightyear/ruby/buzz_lightyear.rb
# Buzz Lightyear Code Challenge
#
# Give all possible values of `x` where `x == x + 2` will evaluate to `true`

start_time = Time.now

[1.0/0, -1.0/0, 1e17].each { |x| p "#{x} evaluates to true" if x == x + 2 }

puts "\nCompleted in #{Time.now - start_time} ms"

File: easy/buzz_lightyear/php/buzz_lightyear.php
<?php

/**
 * Buzz Lightyear Code Challenge
 *
 * Give all possible values of `$x` where `$x == $x + 2;` will evaluate to `true`
 */

$values = array(1e17, PHP_INT_MAX, true);

foreach ($values as $x) {
	if ($x == $x + 2) {
		echo '$x == $x + 2; evaluates to true when $x = ?' . "\n";
		echo "\n";var_export($x);echo "\n";
		var_dump($x == $x + 2);
		echo "\n\n";
	}
}


?>